py/server/deephaven/table.py

```diff
@@ -3996,3 +4009,45 @@ def table_diff(t1: Table, t2: Table, max_diffs: int = 1, floating_comparison: Li
         raise DHError(e, "table diff failed") from e
 
 # endregion
+
+def keyed_transpose(table: Table, aggs: Union[Aggregation, Sequence[Aggregation]],
+                    row_by_cols: Union[str, Sequence[str]], col_by_cols: Union[str, Sequence[str]],
+                    initial_groups: Table = None,
+                    new_column_behavior: NewColumnBehaviorType = NewColumnBehaviorType.FAIL) -> Table:
+    """The keyed_transpose operation takes a source table with a set of aggregations and produces a new table where
+    the columns specified in row_by_columns are the keys used for the aggregation, and the values for the columns
+    specified in column_by_columns are used for the column names. An optional set of initial_groups can be
+    provided to ensure that the output table contains the full set of aggregated columns, even if no data is present yet
+    in the source table.
+
+    Args:
+        table (Table): The source table to transpose.
+        aggs (Union[Aggregation, Sequence[Aggregation]]): The aggregation(s) to apply to the source table.
+        row_by_cols (Union[str, Sequence[str]]): The column(s) to use as row keys in the transposed table.
+        col_by_cols (Union[str, Sequence[str]]): The columns whose values become the new aggregated columns.
+        initial_groups (Table, optional): An optional initial set of groups to ensure all columns are present in the output.
+        new_column_behavior (NewColumnBehaviorType, optional): The behavior when a new column would be added to the table.
+    Returns:
+        a new table
+
+    Raises:
+        DHError
+    """
+    try:
+        j_source_table = unwrap(table)
+        aggs = to_sequence(aggs)
+        row_by_cols = to_sequence(row_by_cols)
+        col_by_cols = to_sequence(col_by_cols)
+        if not row_by_cols and initial_groups:
+            raise ValueError("missing row_by_cols column names when initial_groups is provided.")
```

Comment:
initial groups could be for either row_by or col_by cols?  What is this particular check about?  I don't actually know what the not will mean for these values.